domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -3145,7 +3246,7 @@ public void selectSingle(AsyncResult<SQLConnection> conn, String sql, Handler<As
    * @param replyHandler  The query result or the failure.
    */
   public void selectSingle(String sql, Tuple params, Handler<AsyncResult<Row>> replyHandler) {
-    getSQLConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
+    getSQLReadConnection(conn -> selectSingle(conn, sql, params, closeAndHandleResult(conn, replyHandler)));
```

Comment:
sql can be a writing SQL: `UPDATE foo SET jsonb=$1 WHERE id='123' RETURNING jsonb`

Therefore this change cannot be made in this `selectSingle` method but requires a new `selectReadSingle` method.